### PR TITLE
Device room selector component

### PR DESCRIPTION
**Pull request by atrovato**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- ~~[ ] If your changes affects code, did your write the tests?~~
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- ~~[ ] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)~~
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- ~~[ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)~~
- ~~[ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).~~
- ~~[ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).~~

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Next common component: device room selector.

### CODE REVIEW

**Review by Pierre-Gilles — APPROVED** [review]

File: front/src/components/device/form/DeviceRoomSelector.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { Component } from 'preact';
+import { Text } from 'preact-i18n';
+import { connect } from 'unistore/preact';
+import cx from 'classnames';
+
+import actions from '../../../actions/house';
+import { RequestStatus } from '../../../utils/consts';
+
+@connect('houses,housesGetStatus', actions)
```
Comment (Pierre-Gilles) [inline front/src/components/device/form/DeviceRoomSelector.jsx]: I think there is an issue with the variable name `houses`

As the state is global, it's shared with all other componenents

With have some components that use the same key (`houses`) to store a list of houses without the rooms

It caused some issues in the past, that's why I renamed this key `housesWithRooms` when the room was included

Two ways of fixing this: 

- Rename the key to `housesWithRooms` here
- Stop using global variables (but we are losing the cache)

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]

File: front/src/components/device/form/DeviceRoomSelector.jsx (hunk 1 above)
Comment (Pierre-Gilles) [reply front/src/components/device/form/DeviceRoomSelector.jsx]: Example here : https://github.com/GladysAssistant/Gladys/blob/master/front/src/components/boxs/weather/EditWeatherBox.jsx#L18

Comment (atrovato) [reply front/src/components/device/form/DeviceRoomSelector.jsx]: Nice, thank for this point.
I will rename to `housesWithRooms` and (try to) check all other usages.

**Comment by atrovato** [comment]
I will take it back and split into multiple smaller PRs, to migration integration pages one by one.
Keep tracking on https://github.com/atrovato/Gladys/pull/40

**Pull request closed without merging**